packages/datadog-instrumentations/src/helpers/instrument.js

```diff
@@ -118,3 +119,11 @@ if (semver.satisfies(process.versions.node, '>=16.0.0')) {
     }
   }
 }
+
+exports.bindEmit = function bindEmit (emitter, asyncResource) {
```

Comment:
An emitter should never be bound to a specific context for all events. Only problematic events that other cannot be run in the correct context because we have no access to the surrounding code should be bound. This means it would be specific to the instrumentation and not a generic helper.